### PR TITLE
Added simple Column and Schema classes

### LINKED ISSUE #146: Schema
SqlTable will be responsible for translating the SQL notion of schema to the storage layer's BlockLayout. We'll likely need to bring in Peloton's representation of schema. We also want to add the notion of layout_version to the DataTable level.

### PR DESCRIPTION
**Pull request by mbutrovich**
Fix #146.

Need these since SqlTable will be instantiated with Schema rather BlockLayout. This seems like the minimum functionality we need now, but happy to discuss if I overlooked something.

### CODE REVIEW

File: src/include/catalog/schema.h [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+     * @param oid internal unique identifier for this column
+     */
+    Column(std::string name, const type::TypeId type, const bool nullable, const col_oid_t oid)
+        : name_(std::move(name)),
+          type_(type),
+          attr_size_(type::TypeUtil::GetTypeSize(type_)),
+          nullable_(nullable),
+          inlined_(true),
+          oid_(oid) {
+      if (attr_size_ == 0) {
```
Comment (tli2) [inline src/include/catalog/schema.h]: maybe `#define VARLEN 0`?

File: src/include/catalog/schema.h [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+   private:
+    const std::string name_;
+    const type::TypeId type_;
+    uint8_t attr_size_;
+    const bool nullable_;
+    bool inlined_;
+    const col_oid_t oid_;
+    // TODO(Matt): default value would go here
+    // Value default_;
```
Comment (tli2) [inline src/include/catalog/schema.h]: Should this Column also provide mapping to col_id_t in the storage layer?

File: src/include/common/typedefs.h [hunk 3]
```diff
@@ -214,6 +214,7 @@ using byte = std::byte;
 STRONG_TYPEDEF(timestamp_t, uint64_t);
 STRONG_TYPEDEF(layout_version_t, uint32_t);
 STRONG_TYPEDEF(col_id_t, uint16_t);
+STRONG_TYPEDEF(col_oid_t, uint32_t);
```
Comment (tli2) [inline src/include/common/typedefs.h]: Not in your PR, but this `col_id_t`, `col_oid_t` thing is making me sad.

I will probably later refactor all these strong typedefs into namespaces. (that way we will get `storage::col_id_t` and `catalog::col_oid_t`)

File: src/include/catalog/schema.h (hunk 2 above)
Comment (mbutrovich) [reply src/include/catalog/schema.h]: Possibly! We don't have a catalog yet. I almost think the mapping from col_oit_t to col_id should exist at the SqlTable though, where we're handle stuff like multiple layouts...

File: src/include/common/typedefs.h (hunk 3 above)
Comment (mbutrovich) [reply src/include/common/typedefs.h]: That sounds reasonable.

Comment (mbutrovich) [reply src/include/common/typedefs.h]: Want me to add a todo in my PR?

File: src/include/type/type_util.h [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 namespace terrier::type {
 
+#define VARLEN 0
```
Comment (tli2) [inline src/include/type/type_util.h]: nit: does this belong in type_util? I think it probably belongs in common_def?

**Pull request merged**